projects/common/src/utilities/angular/dynamic-component.service.ts

```diff
@@ -29,4 +33,32 @@ export class DynamicComponentService {
 
     return domPortalOutlet.attach(componentPortal);
   }
+
+  /**
+   * Dynamically create a component and insert it into the view container. Data and services can be provided to the dynamic
+   * component via `providers`
+   *
+   * @returns ComponentRef or undefined if the component could not be created
+   */
+  public attachComponentToViewContainer<T>(opts: DynamicComponentOptions<T>): ComponentRef<T> | undefined {
+    if (!isNil(opts.viewContainerRef) && !isNil(opts.componentClass)) {
+      const componentFactory = this.componentFactoryResolver.resolveComponentFactory(opts.componentClass);
```

Comment:
This should come from the provided injector. See line 24 above. Different child injectors have different components available to them, so using the service level one would only allow access to components available to the root injector.

Reply:
Actually, now that I think about it - think a view container ref provides a number of these things already (injector and element) so it's really just a more specialized form of the above API. We should look to see if we can make the API support both cases, but if it's easier to have two methods, we should at least look to consolidate their implementation.

Reply:
Got it. Updating.